### ISSUE TITLE
Table level unique constraint on columns which also have a foreign key constaint fails

### ISSUE THREAD

**Issue by derkan** [cockroachdb/cockroach#13399]
Is this a question, feature request, or bug report?
> bug report

**QUESTION**

Have you checked our documentation at https://cockroachlabs.com/docs/? If you could not find an answer there, please consider asking your question in our community forum at https://forum.cockroachlabs.com/, as it would benefit other members of our community.

>yes

Prefer live chat? Message our engineers on our Gitter channel at https://gitter.im/cockroachdb/cockroach. 

**FEATURE REQUEST**

1. Please describe the feature you are requesting.

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

3. Provide any additional detail on your proposed use case for this feature.

**BUG REPORT**

1. Please supply the header (i.e. the first few lines) of your most recent
   log file **for each node in your cluster**. On most unix-based systems
   running with defaults, this boils down to the output of

     grep -F '[config]' cockroach-data/logs/cockroach.INFO

   When log files are not available, supply the output of `cockroach version`
   and all flags/environment variables passed to `cockroach start` instead.

2. Please describe the issue you observed:
Column level unique constraint on columns having a Foreign Key constaint fails.
- What did you do?
CockrochDB gives this error at `./cockroach sql` client for `domain_modules`:
```
pq: column "module_id" cannot be used by multiple foreign key constraints
```
```sql

CREATE TABLE modules (
  id          BIGSERIAL    NOT NULL PRIMARY KEY,
  name        VARCHAR(128) NOT NULL UNIQUE
);

CREATE TABLE domains (
  id             BIGSERIAL    NOT NULL PRIMARY KEY,
  domain         VARCHAR(128) NOT NULL UNIQUE
);

CREATE TABLE domain_modules (
  id         BIGSERIAL    NOT NULL PRIMARY KEY,
  domain_id  BIGINT       NOT NULL,
  module_id  BIGINT       NOT NULL,
  CONSTRAINT domain_modules_domain_id_fk FOREIGN KEY (domain_id) REFERENCES domains (id),
  CONSTRAINT domain_modules_module_id_fk FOREIGN KEY (module_id) REFERENCES modules (id),
  CONSTRAINT domain_modules_uq UNIQUE (domain_id, module_id)
);
```
- What did you expect to see?

```
pq: column "module_id" cannot be used by multiple foreign key constraints
```
- What did you see instead?
Docs says Column level unique constrain is supported.

**Comment by dt** [comment]
https://github.com/cockroachdb/cockroach/issues/8345 strikes again.

**Comment by dt** [comment]
Fixed in #13439
